src/mixins/java/org/spongepowered/common/mixin/core/world/entity/EntityMixin.java

```diff
@@ -240,6 +241,32 @@
         this.impl$isConstructing = false;
     }
 
+    @Override
+    public boolean bridge$setPosition(final Vector3d position) {
+        if (this.removed) {
+            return false;
+        }
+
+        try (final CauseStackManager.StackFrame frame = PhaseTracker.getCauseStackManager().pushCauseFrame()) {
+            frame.pushCause(SpongeCommon.getActivePlugin());
+            frame.addContext(EventContextKeys.MOVEMENT_TYPE, MovementTypes.PLUGIN);
+
+            final net.minecraft.world.phys.Vec3 originalPosition = this.shadow$position();
+            Vector3d destinationPosition = position;
+            if (ShouldFire.MOVE_ENTITY_EVENT) {
+                final MoveEntityEvent event = SpongeEventFactory.createMoveEntityEvent(frame.currentCause(),
+                    (org.spongepowered.api.entity.Entity) this, VecHelper.toVector3d(this.shadow$position()), position, position);
+                if (SpongeCommon.postEvent(event)) {
+                    return false;
+                }
+                destinationPosition = event.destinationPosition();
+            }
+            this.shadow$teleportToWithTicket(destinationPosition.x(), destinationPosition.y(), destinationPosition.z());
```

Comment:
`teleportToWithTicket` has been overwritten to fire the event anyway. Shouldn't you just call that and not fire the event here?